ios/FluentUI/Tab Bar/TabBarItemView.swift

```diff
@@ -173,6 +188,23 @@ class TabBarItemView: UIControl {
         }
     }
 
+    @objc private func isUnreadValueDidChange() {
+        isUnreadDotVisible = item.isUnreadDotVisible
+        updateBadgeView()
+        updateAccessibilityLabel()
+        setNeedsLayout()
+    }
+
+    private var isUnreadDotVisible: Bool = false {
+        didSet {
+            if oldValue != isUnreadDotVisible {
+                if isUnreadDotVisible {
+                    updateBadgeView()
+                }
+            }
+        }
```

Comment:
is all your logic of didset{} needed? now that you have unified updateBadgeView() it should just be handle by isUnreadValueDidChange and badgevaluedidchange because once the isunreaddot is set to false, you already triggered mask to be nil.